src/sidebar/components/tag-editor.js

```diff
@@ -10,19 +10,46 @@ import { normalizeKeyName } from '../../shared/browser-compatibility-utils';
 import SvgIcon from '../../shared/components/svg-icon';
 import useElementShouldClose from './hooks/use-element-should-close';
 
+/**
+ * @template T
+ * @typedef {import("preact/hooks").StateUpdater<T>} StateUpdater
+ */
+
+/**
+ * @template T
+ * @typedef {import("preact/hooks").Ref<T>} Ref
+ */
+
+/**
+ * @template T
+ * @typedef {import("preact/hooks").PropRef<T>} PropRef
```

Comment:
The `PropRef` name that "preact/hooks" uses is a bit confusing. From looking at the `index.d.ts` file in the "preact/hooks" package it appears that the difference is that `PropRef` is nullable whereas `Ref` is not.

Reply:
indeed, 

this works
```
/** @type {Ref<HTMLInputElement>} */
  const inputEl = useRef(new HTMLInputElement());
```

```
this does not work
/** @type {Ref<HTMLInputElement>} */
  const inputEl = useRef();
```

which is why I switch it to a PropRef, but the PropRef name is very poor and not obvious at all. "NullableRef" would be better.

Reply:
Thinking more on this
```
/** @type {Ref<HTMLInputElement|null>} */
  const inputEl = useRef(null);

```
also could work, I'm just not sure what direction we want to proceed with.

Reply:
Can I request that we use this convention for non-nullable refs in this PR:

```js
const inputEl = useRef(/** @type {HTMLInputElement|null} */ (null));
```

With this, there is no need to import `Ref`.